src/server/game/Scripting/ScriptReloadMgr.cpp

```diff
@@ -67,6 +67,9 @@ namespace fs = boost::filesystem;
 #ifdef _WIN32
     #include <windows.h>
     #define HOTSWAP_PLATFORM_REQUIRES_CACHING
+#elif __APPLE__
+    #include <dlfcn.h>
+    #define HOTSWAP_PLATFORM_REQUIRES_CACHING
```

Comment:
Is caching of binaries actually required on OSX?
The BSD kernel should behave similar to Linux not locking the loaded shared libraries on usage.